docs/source/array-best-practices.rst

```diff
@@ -0,0 +1,119 @@
+
+
+Orient your chunks
+------------------
+
+When reading data you should align your chunks with your storage format.
+Most array storage formats store data in chunks themselves.  If your Dask array
+chunks aren't multiples of these chunk shapes then you will have to read the
+same data repeatedly, which can be very expensive.  Note though that often
+storage formats choose chunk sizes that are much smaller than is ideal for
+Dask, closer to 1MB than 100MB.  In these cases you should choose a Dask chunk
+size that aligns with the storage chunk size and that every Dask chunk
+dimension is a multiple of the storage chunk dimension.
+
+So for example if we have an HDF file that has chunks of size ``(128, 64)``, we
+might choose a chunk shape of ``(1280, 6400)``.
+
+.. code-block:: python
+
+   >>> import h5py
+   >>> storage = h5py.File('myfile.hdf5')['x']
+   >>> storage.chunks
+   (128, 64)
+
+   >>> import dask.array as da
+   >>> x = da.from_array(storage, chunks=(1280, 6400))
+
+Note that if you provide ``chunks='auto'`` then Dask Array will look for a
+``.chunks`` attribute and use that to provide a good chunking.
+
+
+Avoid Oversubscribing Threads
+-----------------------------
+
+By default Dask will run as many concurrent tasks as you have logical cores.
+It assumes that each task will consume about one core.  However, many
+array-computing libraries are themselves multi-threaded, which can cause
+contention and low performance.  In particular the BLAS/LAPACK libraries that
+back most of NumPy's linear algebra routines are often multi-threaded, and need
+to be told to use only one thread explicitly.  You can do this with the
+following environment variables (using bash ``export`` command below, but this
+may vary depending on your operating system).
+
+.. code-block::
+
+   export OMP_NUM_THREADS=1
+   export MKL_NUM_THREADS=1
+
+You need to run this before you start your Python process for it to take effect.
+
+
+Consider Xarray
+---------------
+
+The `Xarray <http://xarray.pydata.org/en/stable/>`_ package wraps around Dask
+Array, and so offers the same scalability, but also adds convenience when
+dealing with complex datasets.  In particular Xarray can help with the
+following:
+
+1.  Manage multiple arrays together as a consistent dataset
+2.  Read from a stack of HDF or NetCDF files at once
+3.  Switch between Dask Array and NumPy with a consistent API
+
+Xarray is used in wide range of fields, including physics, astronomy, geoscience,
+bioinformatics, engineering, finance, and deep learning.
+Xarray also has a thriving user community that is good at providing support.
+
+
+Build your own Operations
+-------------------------
+
+Often we want to perform computations for which there is no exact function
+in Dask Array.  In these cases we may be able to use some of the more generic
+functions to build our own.  These include:
+
+.. currentmodule:: dask.array
+
+.. autosummary::
+   map_blocks
+   reduction
+   map_overlap
```

Comment:
Minor suggestion would be to group `map_blocks` and `map_overlap` together. Though no strong feelings if that is opposed.

Note: Sorry for not adding a suggested change. GitHub had trouble doing this correctly here.

Reply:
Yeah, the single-line restriction is limiting.  I'll make the change and push up normally.